src/iface/ethernet.rs

```diff
@@ -165,26 +165,26 @@ impl<'a, 'b, 'c, DeviceT: Device + 'a> Interface<'a, 'b, 'c, DeviceT> {
         let mut limits = self.device.limits();
         limits.max_transmission_unit -= EthernetFrame::<&[u8]>::header_len();
 
-        for socket in sockets.iter_mut() {
+        for mut socket in sockets.iter_mut() {
             let mut device_result = Ok(());
             let socket_result =
-                match socket {
-                    &mut Socket::Raw(ref mut socket) =>
+                match *(&mut *socket) {
```

Comment:
*squint* what... is this thing

Reply:
Looks like I've got carried away a little.